main/test/ca/uwaterloo/flix/language/phase/TestParser.scala

```diff
@@ -731,6 +731,87 @@ class TestParser extends FunSuite with TestUtils {
     run(input)
   }
 
+  test("Expression.Array.01") {
+    val input = "def f(): Unit = [||]"
+    run(input, core = false)
+
+  }
+
+  test("Expression.ArrayLit.02") {
+    val input = "def f(): Int = [|42|] "
+    run(input, core = false)
+  }
+
+  test("Expression.ArrayLit.03") {
+    val input = "def f(): Int = [|42, 21|]"
+    run(input, core = false)
+  }
+
+  test("Expression.ArrayLit.03") {
+    val input = "def f(): Int = [|42, x, 21|]"
+    run(input, core = false)
+  }
+
+  test("Expression.ArrayLit.04") {
+    val input = "def f(): Int = [|42, x, 21|]"
+    run(input, core = false)
+  }
+
+  test("Expression.ArrayLit.05") {
+    val input = "def f(): Int = [|42, x, x, 21|]"
+    run(input, core = false)
+  }
+
+  test("Expression.ArrayNew.01") {
+    val input = "def f(): Unit = [||]"
+    run(input, core = false )
+  }
+
+  test("Expression.ArrayNew.02") {
+    val input = "def f(): Unit = [|0;0|]"
+    run(input, core = false )
+  }
+
+  test("Expression.ArrayNew.03") {
+    val input = "def f(): Int = [|0;1|]"
+    run(input, core = false )
+  }
+
+  test("Expression.ArrayNew.04") {
+    val input = "def f(): Int [|0;2|]"
+    run(input, core = false )
+  }
+
+  test("Expression.ArrayNew.05") {
+    val input = "def f(): Int [|0;3|]"
+    run(input, core = false )
+  }
```

Comment:
This is the kind of systematic testing I want to see, i.e. you are testing ArrayNew and so you try with length 0..3,
next you need to repeat it for all the types mentioned below, e.g. booleans, floats, etc. etc.